### PR TITLE
feat: update chart to enable crs and topology features

### LINKED ISSUE #130: Update the chart to enable the CRS & Topology features
Update the chart so that when the specific 1.4.x version of core CAPI is being installed we enable the following features:

- CRS
- Topology

This will give a better experience for users when they get started.

### PR DESCRIPTION
**Pull request by salasberryfin**
**What this PR does / why we need it**:

This PR updates the chart so that  when it deploys the CoreProvider it also sets the environment variables for the features:

- CRS
- Topology

These features are currently set by [manually creating a secret](https://docs.rancher-turtles.com/docs/getting-started/install_turtles_operator) with the desired values for the environment variables `CLUSTER_TOPOLOGY` and `EXP_CLUSTER_RESOURCE_SET`. This secret will now be created as part of the chart installation and both features will default to `true`. These can otherwise be disabled passing the parameters to Helm:

`--set cluster-api-operator.cluster-api.secretConfig.name="user-secret-name"`
`--set cluster-api-operator.cluster-api.secretConfig.namespace="user-secret-namespace`

When both these parameters are provided, the secret is expected to exist in the namespace and the user must manage its contents, including (but not limited to) `CLUSTER_TOPOLOGY` and `EXP_CLUSTER_RESOURCE_SET`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #130 

**Special notes for your reviewer**:

README is updated with the new values and a new issue is created to track `rancher-turtles-docs` updates rancher-sandbox/rancher-turtles-docs#27.

**Checklist**:

- [x] squashed commits into logical changes
- [x] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

File: test/e2e/helpers_test.go [hunk 1]
```diff
@@ -30,9 +30,6 @@ var (
 	//go:embed resources/testdata/fleet-capi-test.yaml
 	fleetCAPITestdata []byte
 
-	//go:embed resources/config/capi-providers-secret.yaml
-	capiProvidersSecret []byte
```
Comment (Danil-Grigorev) [inline test/e2e/helpers_test.go:34]: Why is this removed? This secret is a “patch” on top of existing resource as well, and is used for providing additional env depending on type of test. Like in e2e for azure PR, it adds all azure env secrets.

Comment (salasberryfin) [reply test/e2e/helpers_test.go:34]: It was removed just because I did not see it being used other than in the `Adding CAPI variables secret` section of [initRancherTurtles](https://github.com/rancher-sandbox/rancher-turtles/blob/ed9b78bb4cd64dab12dee4feb5dac19cab538ced/test/e2e/e2e_suite_test.go#L249), which is no longer required, but I have no problem in keeping it if it's needed.

File: charts/rancher-turtles/templates/core-provider.yaml [hunk 2]
```diff
@@ -68,4 +64,17 @@ data:
       - apiGroups: ["rke-machine.cattle.io"]
         resources: ["*"]
         verbs: ["*"]
+---
```
Comment (Danil-Grigorev) [inline charts/rancher-turtles/templates/core-provider.yaml:67]: What if a user would want to use his own secret, already created in the cluster?

File: test/e2e/helpers_test.go (hunk 1 above)
Comment (Danil-Grigorev) [reply test/e2e/helpers_test.go:34]: For this to be no longer required a secret content should be fully customizable. Right now you can only set 2 feature flags.

**Review by Danil-Grigorev — CHANGES_REQUESTED** [review]

File: charts/rancher-turtles/templates/core-provider.yaml (hunk 2 above)
Comment (salasberryfin) [reply charts/rancher-turtles/templates/core-provider.yaml:67]: I will add a flag to use an existing secret. The default behavior would be to create the pre-configured secret but the user can pass this parameter to avoid this.

File: test/e2e/helpers_test.go (hunk 1 above)
Comment (salasberryfin) [reply test/e2e/helpers_test.go:34]: If we give the user the possibility to use their own secret, do we still need to let them customize the content of the "default" secret?

File: charts/rancher-turtles/templates/core-provider.yaml (hunk 2 above)
Comment (Danil-Grigorev) [reply charts/rancher-turtles/templates/core-provider.yaml:67]: The simplest way how to achieve the goal - to not create anything if name and namespace is not set.

Comment (salasberryfin) [reply charts/rancher-turtles/templates/core-provider.yaml:67]: The initial idea was to use an opt-out strategy rather than opt-in, so the secret would be created by default and the values for these two features set to true. They would be able to explicitly turn them off.

File: test/e2e/helpers_test.go (hunk 1 above)
Comment (Danil-Grigorev) [reply test/e2e/helpers_test.go:34]: This functionality used in e2e tests. There is a PR implementing test case for [azure](https://github.com/rancher-sandbox/rancher-turtles/pull/133), it was opened a couple days prior to that. This is where the secret is used and extended.

**Comment by salasberryfin** [comment]
I would be more in favor of @alexander-demicev's proposal. I think this is a simpler solution and gives the user the freedom to manage their own secret if they want to do so, which I would say covers both point 2 and 3 in @Danil-Grigorev's list of use cases.

The deletion of the secret patching functionality was just an initial clean-up proposal which I'll undo. This could use a user-managed secret (current status) with any custom environment variables/feature flags.

**Review by Danil-Grigorev — APPROVED** [review]

**Review by richardcase — DISMISSED** [review]

**Comment by salasberryfin** [comment]
There was a condition missing in `core-provider.yaml` when setting `CoreProvider` https://github.com/rancher-sandbox/rancher-turtles/blob/d73fd08d01e71aa21ecdecfa23d0269f94cfc3d6/charts/rancher-turtles/templates/core-provider.yaml#L24 It should now be fixed

**Review by Danil-Grigorev — APPROVED** [review]

**Review by alexander-demicev — APPROVED** [review]
